upgrade/util/util.go

```diff
@@ -302,7 +302,7 @@ func checkEngineUpgradePath(namespace string, lhClient lhclientset.Interface, ne
 		return err
 	}
 
-	if len(engineImageVersions) == 0 {
+	if len(engineImageVersions) == 0 || !enableUpgradeVersionCheck {
```

Comment:
You can skip `!enableUpgradeVersionCheck` at the beginning of the method, but do it after querying the engine images.

Reply:
Moved codes
```golang
if !enableUpgradeVersionCheck {
	if semver.Compare(lhCurrentVersion, "v1.4.0") < 0 {
		return fmt.Errorf("failed to skip the upgrade check for %s to upgrade to %s, as it only supports the current version larger than v1.4.0", lhCurrentVersion, meta.Version)
	}
	logrus.Warnf(skipUpgradeVersionCheckMsg, lhCurrentVersion, meta.Version)
	eventRecorder.Eventf(&corev1.ObjectReference{Namespace: namespace, Name: "longhorn-upgrade"}, corev1.EventTypeWarning, constant.EventReasonUpgrade, skipUpgradeVersionCheckMsg, lhCurrentVersion, meta.Version)
	return nil
}
```
from checkLHUpgradePath to the beginning of CheckUpgradePath.
And removed passing variables to downstream methods.
Thanks.